### PR TITLE
Fix for #644 and another NPE

### PR DESCRIPTION
**Pull request by NickAragua**
Just what it says.

For some reason, the quirks collection includes a null. Meanwhile, addEquipment() invokes addComponent() on a null compositeTechLevel.

### CODE REVIEW

**Comment by NickAragua** [comment]
Looks like there's going to be about 20 more of these scattered throughout the code. Maybe a better approach would be to ensure that calls to IOptionGroup.getOptions() don't contain null elements.

File: megamek/src/megamek/common/Entity.java [hunk 1]
```diff
@@ -3395,9 +3395,12 @@ public void addEquipment(Mounted mounted, int loc, boolean rearMounted)
             throws LocationFullException {
         mounted.setLocation(loc, rearMounted);
         equipmentList.add(mounted);
-        compositeTechLevel.addComponent(mounted.getType());
-        if (mounted.isArmored()) {
-            compositeTechLevel.addComponent(TA_ARMORED_COMPONENT);
+
```
Comment (neoancient) [inline megamek/src/megamek/common/Entity.java]: 1. These issues are unrelated and should not be in the same PR.
2. The problem here is that `compositeTechLevel` should never be null. Skipping this section of code if it has an illegal value does not fix the problem. It hides it.

Comment (NickAragua) [reply megamek/src/megamek/common/Entity.java]: I'll implement the "right" fix for these issues in another PR then. 

For my education:

At what point is the quirks list initialized? Why does it get continuously reloaded in the middle of a tactical game?

At what point during a tactical game is the compositeTechLevel information of an on/offboard unit used?

Comment (pheonixstorm) [reply megamek/src/megamek/common/Entity.java]: Entity ln822 `quirks.initialize();`
I didn't find any other areas that call initialize in Quirks.java outside of Entity and AbstractOptions

**Comment by arlith** [comment]
I think that the quirk fix is in the same class as the `compositeTechLevel` fix; you're not actually fixing the problem but just hiding it.  This is a crash I've seen often (and never really done anything about) that is generally caused by loading a saved game across versions, where the new version that has changes related to quirks (additions, removals, name changes).

In this case, ignoring the null may be a viable option, but you should clearly understand the implications of this, and not just pushing the problem down to a crash later.

**Comment by dericpage** [comment]
I concur with Arlith.  At the very least, if there's a null element in the quirks list, a log entry should be made (probably at the Warning level) listing the name of the group that contained the null.

**Comment by NickAragua** [comment]
Closing for now, will resubmit with different set of fixes.
